plugin/storage/es/factory.go

```diff
@@ -50,6 +52,7 @@ type Factory struct {
 
 	metricsFactory metrics.Factory
 	logger         *zap.Logger
+	tracer         trace.TracerProvider
```

Comment:
@yurishkuro I am still facing an error here quite scary. Maybe I am missed out here.
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
```

Reply:
look at full stack trace in the panic log and find where nil is being passed.

Reply:
I drilled down to this `trace, err := r.reader.GetTrace(context.Background(), model.NewTraceID(0, 1))`

Reply:
sorry you mean e.g `jaeger-all-in-one` main? Then yes.

Reply:
I don't see it in the code

Reply:
I tried this out in my local test. I still failed to get the traces. I don't understand this `github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanReader).GetTrace(0xc000000cc0, {0xd64a90?, 0xc0000440d8?}, {0xc00007f480?, 0x20?})` I guess this has pointer to address which is not nil.

Reply:
Please push changes to pr so that I can reproduce. Is this when running tests or main binaries? (Also, you can avoid this back and forth by posting full stack trace)

Reply:
in the spanreader_test, I see many tests are using `withSpanReader()`, which creates a reader like this:
```
		reader: NewSpanReader(SpanReaderParams{
			Client:            client,
			Logger:            zap.NewNop(),
			MaxSpanAge:        0,
			IndexPrefix:       "",
			TagDotReplacement: "@",
			MaxDocCount:       defaultMaxDocCount,
		}),
```
The tracer is not initialized there at all, so not surprising you get a null pointer panic